### ISSUE TITLE
set_epoch for DistributedSampler

### ISSUE THREAD

**Issue by ArchNew** [HRNet/DEKR#13]
**Describe the bug**
PyTorch example suggests the use set_epoch function for DistributedSampler class before each epoch start. I could not find anywhere in your code.

https://github.com/pytorch/examples/blob/master/imagenet/main.py
Line 232-234

As can be seen from the DistributedSampler class code (https://github.com/pytorch/pytorch/blob/master/torch/utils/data/distributed.py), the set_epoch function is required to set the seed for each iter function call.

Can you confirm if this function has been called on DistributedSampler (for training dataset) at some point in your code?

Copyright Claim: I ask the same question as @ananyahjha93 did. Hence I copied and slight modified his post here: https://github.com/PyTorchLightning/pytorch-lightning/issues/224#issue-493778958

**Comment by Gengzigang** [comment]
Hi, you can find the DistributedSampler in the lib/dataset/build.py.

**Comment by Gengzigang** [comment]
I am very sorry for misunderstanding you. I indeed did not call this function and I will fix it and test the results.

**Comment by ArchNew** [comment]
> I am very sorry for misunderstanding you. I indeed did not call this function and I will fix it and test the results.'

I'm sure you can get better results than those reported in your paper. :D
